### PR TITLE
Fix/carlacda sim adapter xmlrpc api

### PR DESCRIPTION
**Pull request by paulbourelly999**
# PR Details
## Description

Updated carla-cdasim-adapter to have simpler methods for creating sensors and returning detections.
## Related Jira Key

[CDAR-209](https://usdot-carma.atlassian.net/browse/CDAR-209)
## Motivation and Context
Improve CarlaCDASimAdapter usability

## How Has This Been Tested?
Tested using java test xmlrpc client and docker image running carlasim/carla:0.9.10. 
Also tested docker image build process and deployment and works

## Types of changes

- [x] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

[CDAR-209]: https://usdot-carma.atlassian.net/browse/CDAR-209?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by paulbourelly999** [comment]
I should probably update the documentation to reflect the API changes I made. Will push a commit as soon I make those changes.

File: sensorlib/doc/CarlaCDASimAdapter.md [hunk 1]
```diff
@@ -69,24 +69,25 @@ optional arguments:
                         XML-RPC server host. (default: "localhost")
   --xmlrpc-server-port XMLRPC_SERVER_PORT
                         XML-RPC server port. (default: 8000)
+  --sensor-config-file
```
Comment (davidgayman) [inline sensorlib/doc/CarlaCDASimAdapter.md:72]: Looks perfect.

**Review by davidgayman — COMMENTED** [review]
Hey Paul what do you think will be the best approach for handling parent actor ID?

File: sensorlib/src/CarlaCDASimAdapter.py [hunk 2]
```diff
@@ -50,20 +58,17 @@ def start_xml_rpc_server(self, xmlrpc_server_host, xmlrpc_server_port, blocking=
...
                                                  infrastructure_id, sensor_id,
-                                                 sensor_position, sensor_rotation, parent_actor_id):
-        sensor_config = SimulatedSensorUtils.load_config_from_file(sensor_config_file)
-        noise_model_config = SimulatedSensorUtils.load_config_from_file(noise_model_config_file)
-
-        simulated_sensor = self.__api.create_simulated_semantic_lidar_sensor(sensor_config["simulated_sensor"],
-                                                                             sensor_config["lidar_sensor"],
-                                                                             noise_model_config,
-                                                                             detection_cycle_delay_seconds,
+                                                 sensor_position, sensor_rotation):
```
[the remark this answers is not in the document]
Comment (paulbourelly999) [reply sensorlib/src/CarlaCDASimAdapter.py:63]: Please let me know if your experience with CARLA and the semantic lidar sensors is different in any way from what I represented here. Mainly about parent actor ID not having any functional impact on sensor when the parent actor is static.

[the remark this answers is not in the document]
Comment (davidgayman) [reply sensorlib/src/CarlaCDASimAdapter.py:63]: Got it. From my experience the sensor is able to be spawned without a parent actor. The only drawback to removing parent id is we will be unable to attach it to a moving object such as a vehicle. If that has been given consideration and we want to remove that capability.

**Review by davidgayman — DISMISSED** [review]

**Review by dg-developer — APPROVED** [review]

**Review by davidgayman — APPROVED** [review]

**Pull request merged**